### ISSUE TITLE
Another false positive SING_SINGLETON_IMPLEMENTS_SERIALIZABLE when class is not a singleton (e.g. protobuf)

### ISSUE THREAD

**Issue by cpfeiffer** [spotbugs/spotbugs#2985]
https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/MultipleInstantiationsOfSingletons.java#L218 lists the criteria for recognizing a class as a Singleton. While already quite sophisticated, it wrongly recognizes some non-singleton classes as singleton. E.g. protobuf-generated classes that contain a builder to create an instance like this (`GeneratedMessageV3` is `Serializable`).

Here's a manually simplified version of such a class generated by the `protoc` compiler:

```java
public final class Person extends
    com.google.protobuf.GeneratedMessageV3 implements PersonOrBuilder {
  private static final long serialVersionUID = 0L;

  private static final Person DEFAULT_INSTANCE;
  static {
    DEFAULT_INSTANCE = new Person();
  }

  public static Person getDefaultInstance() {
    return DEFAULT_INSTANCE;
  }

  private Person(com.google.protobuf.GeneratedMessageV3.Builder<?> builder) {
    super(builder);
  }
  private Person() {
    name_ = com.google.protobuf.LazyStringArrayList.emptyList();
  }

  @java.lang.Override
  @SuppressWarnings({"unused"})
  protected java.lang.Object newInstance( <--- this method should already be an indicator that it is not a singleton
      UnusedPrivateParameter unused) {
    return new Person();
  }

  public static Builder newBuilder() {
    return DEFAULT_INSTANCE.toBuilder();
  }

  public static final class Builder extends
      com.google.protobuf.GeneratedMessageV3.Builder<Builder> implements PersonOrBuilder {
    public Person build() {
      return new Person(this);
    }
  }
}
```

**Comment by PhilippWendler** [comment]
We also have several classes with a public constructor and some default instance in a static field that are wrongly classified as singletons.
